src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -189,6 +189,8 @@
 
       <EmccInitialHeapSize Condition="'$(EmccInitialHeapSize)' == ''">$(EmccTotalMemory)</EmccInitialHeapSize>
       <EmccInitialHeapSize Condition="'$(EmccInitialHeapSize)' == ''">536870912</EmccInitialHeapSize>
+      <WasmAllowUndefinedSymbols Condition="'$(WasmBuildingForNestedPublish)' == 'true'">false</WasmAllowUndefinedSymbols>
```

Comment:
I would prefer to flip the default and warn about missing symbols. Ideally the error message would also mention how to suppress it.

Reply:
The error message comes from the `emcc`.
Do you have any better idea other than parsing the text output?

Reply:
adding msbuild info message triggered by regex should do no harm, even if it was false positive

Reply:
re:emcc-warnings-errors, I'm already working on this locally.

re:default, this would change the current behavior. I think the default should be `true`, so the current behavior is retained. And the warning message can be surfaced specifically with a custom `WASM*` warning code which would allow the user to disable it.

Reply:
It might be useful to try building some sqlite, skiasharp etc like projects to check what undefined symbols we get.

Reply:
Having undefined symbols that don't get linked out currently stops the build. They are treated as errors.

> And the warning message can be surfaced specifically with a custom `WASM*` warning code which would allow the user to disable it.

Having undefined symbols that don't get linked out currently stops the build. They are treated as errors. It is caused by the `ERROR_ON_UNDEFINED_SYMBOLS=1`. We could use instead `WARN_ON_UNDEFINED_SYMBOLS=1` and the wasm-ld will than produces warnings, but it has a side effect that emcc will use arg `--import-undefined` instead of `--allow-undefined-file`

Reply:
adopting https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/interop.md would give us additional information on what to consider an error and what to treat like an import

Reply:
> It might be useful to try building some sqlite, skiasharp etc like projects to check what undefined symbols we get.

For both sqlite and skiasharp, I got no errors about undefined symbols.

Reply:
@radical As sqlite and skiasharp both link without any errors. Do you have some suggestions for other libraries to test? Do you think we should still have a default value `true`?

Reply:
> Ideally the error message would also mention how to suppress it.

This part makes sense only when the mechanism is "opt-out".

Reply:
> I would prefer to flip the default and warn about missing symbols. Ideally the error message would also mention how to suppress it.

Flipping the default value